### DISCUSSION TITLE
microshift ip

### DISCUSSION THREAD

**Discussion by burrsutter** [openshift/microshift#601] [Q&A]
How to gain access to the ip address or what is the trick with gaining access to the Service?

My minikube script:

kubectl create namespace mystuff
kubectl config set-context --current --namespace=mystuff
kubectl create deployment myapp --image=quay.io/rhdevelopers/quarkus-demo:v1
kubectl expose deployment myapp --port=8080 --type=LoadBalancer

IP=$(minikube ip)
PORT=$(kubectl get service/myapp -o jsonpath="{.spec.ports[*].nodePort}")

curl $IP:$PORT

and 

kubectl get services
NAME    TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
myapp   LoadBalancer   10.43.37.209   <pending>     8080:32385/TCP   4m54s

**Comment by cooktheryan** [comment]
@mangelajo can hopefully provide some information around the usage of the OpenShift router. 

NodePort is available out of the box. 

MetalLB  https://microshift.io/docs/user-documentation/how-tos/example-usage/ can be configured as well for LB services

**Reply by burrsutter** [reply]
I am good with whatever the out-of-the-box solution is for exposing a Service.  I did not try an Ingres to see what that might yield.  Historically I use nodeport and loadbalancer which minikube exposed automatically.  minikube service --url <service-name>, there is also minikube tunnel which I dislike because it blocks a process on my machine (and I have to open yet another terminal).

**Comment by mangelajo** [comment]
you can use either NodePort, or a route/ingress out of the box, for example:

```
oc -n default create deployment nginx --image=nginxinc/nginx-unprivileged:stable-alpine
oc -n default expose deployment nginx --port=8080
oc expose service/nginx -n default
oc expose service/nginx -n default --hostname=my-nginx.local

curl http://my-nginx.local
```

Please note that for the mDNS resolver to work in the lan, the hostname of the device where MicroShift is installed must also be .local (we have a bug to be fixed), otherwise the mDNS responder isn't enabled.

You could also use non-mdns names, and target the IP of the device on port 80 / 443 with the right hostname.

**Reply by jiridanek** [reply]
Bug is https://github.com/redhat-et/microshift/issues/563?
